### PR TITLE
BL-932 Report bad generated PDF

### PR DESCRIPTION
**Pull request by JohnThomson**

### CODE REVIEW

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-debug/97/
Build succeeded

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-debug/98/
Build succeeded

File: src/BloomExe/Publish/PdfMaker.cs [hunk 1]
```diff
@@ -71,10 +73,67 @@ public void MakePdf(string inputHtmlPath, string outputPdfPath, string paperSize
...
+				// This is characteristic of BL-932, where Gecko29 fails to make a valid PDF, typically
+				// because the user has embedded a really huge image, something like 4000 pixels wide.
+				// We think it could also happen with a very long book or if the user is short of memory.
+				// The resulting corruption of the PDF file takes the form of a syntax error in an embedded
+				// object so that the parser finds an empty string where it expected a 'generationNumber'
+				// (currently line 106 of Parser.cs). This exception is swallowed but leads to an empty
+				// externalIDs dictionary in PdfImportedObjectTable, and eventually a new exception trying
+				// to look up an object ID at line 121 of that class. We catch that exception here and
+				// suggest possible actions the user can take until we find a better solution.
+				MessageBox.Show(
```
Comment (hatton) [inline src/BloomExe/Publish/PdfMaker.cs]: If instead you use
     Palaso.Reporting.ErrorReport.NotifyUserOfProblem(error, msg),

Then a user would have buttons to report the problem to us.

File: src/BloomExe/Publish/PdfMaker.cs [hunk 2]
```diff
@@ -71,10 +73,67 @@ public void MakePdf(string inputHtmlPath, string outputPdfPath, string paperSize
...
+					LocalizationManager.GetString("PdfMaker.BadPdf", "Bloom unexpectedly failed to create a valid PDF version of this document. The developers are trying to fix this problem. In the meantime, here are some things to try:")
```
Comment (hatton) [inline src/BloomExe/Publish/PdfMaker.cs]: Is it possible to *expectedly* fail?  :-) How about "Bloom had a problem making a PDF of this book. You may need technical help or to contact the developers. But here are some things you can try:"

**Comment by hatton** [comment]
Looks good otherwise. Thanks for doing this.

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-debug/99/
Build succeeded

**Pull request merged**